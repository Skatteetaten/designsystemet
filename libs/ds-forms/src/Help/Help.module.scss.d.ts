declare const classNames: {
  readonly description: 'description';
  readonly descriptionMarginTop: 'descriptionMarginTop';
  readonly helpButton: 'helpButton';
  readonly helpBox: 'helpBox';
  readonly helpBoxText: 'helpBoxText';
  readonly helpBoxCloseButton: 'helpBoxCloseButton';
  readonly srOnly: 'srOnly';
};
export default classNames;
