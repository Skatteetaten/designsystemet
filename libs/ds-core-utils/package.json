{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.1.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14"
  },
  "scripts": {}
}
