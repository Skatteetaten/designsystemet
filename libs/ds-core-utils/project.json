{
  "name": "ds-core-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-core-utils/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-core-utils",
        "tsConfig": "libs/ds-core-utils/tsconfig.lib.json",
        "project": "libs/ds-core-utils/package.json",
        "entryFile": "libs/ds-core-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-core-utils/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-core-utils/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-core-utils"],
      "options": {
        "jestConfig": "libs/ds-core-utils/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-core-utils/**/*.css",
          "libs/ds-core-utils/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-core-utils"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-core-utils {args.switches}"
      }
    }
  }
}
