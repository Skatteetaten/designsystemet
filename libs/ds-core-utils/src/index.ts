export * from './SkatteetatenLogo/SkatteetatenLogo';
export * from './base-props.types';
export * from './defaults';
export * from './ExternalLayout/ExternalLayout';
export * from './InternalLayout/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
export * from './hooks/hooks';
export * from './formatter/formatter';
