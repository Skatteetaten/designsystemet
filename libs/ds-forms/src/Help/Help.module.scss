@use 'libs/ds-core-utils/src/styles';

@keyframes animationhelpbox {
  from {
    opacity: 0;
    pointer-events: none;
    transform: translate3d(0, -20px, 0);
  }

  to {
    opacity: 1;
    pointer-events: auto;
    transform: translate3d(0, 0, 0);
  }
}

.description {
  color: var(--palette-graphite-70);
  font-size: var(--font-size-m);
  line-height: var(--semantic-line-height-input);
  overflow-wrap: anywhere;

  &MarginTop {
    margin-top: var(--spacing-xs);
  }
}

.helpButton {
  margin-left: var(--spacing-s);
}

.helpBox {
  display: flex;
  width: fit-content;
  box-sizing: border-box;
  border: 2px solid var(--palette-forest-30);
  margin-top: var(--spacing-xs);
  animation-duration: 0.3s;
  animation-fill-mode: both;
  animation-name: animationhelpbox;
  animation-timing-function: cubic-bezier(0.1, 0.9, 0.2, 1);
  background-color: var(--palette-forest-10);
  overflow-wrap: anywhere;

  &Text {
    flex-grow: 1;
    padding: var(--spacing-xs) var(--spacing-s);
    color: var(--semantic-page-foreground);
    font-size: var(--font-size-m);
    line-height: var(--semantic-line-height-default);
  }

  &CloseButton {
    align-self: start;
    margin: var(--spacing-xs) var(--spacing-s) var(--spacing-xs) 0;

    &:not(:active) > svg {
      fill: var(--palette-graphite-100);
    }
  }
}

.srOnly {
  @include styles.sr-only;
}
