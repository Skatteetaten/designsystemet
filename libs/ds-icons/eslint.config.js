import nxPlugin from '@nx/eslint-plugin';

import baseConfig from '../../eslint.config.js';

export default [
  ...baseConfig,
  ...nxPlugin.configs['flat/react'],
  {
    ignores: ['!**/*'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
];
