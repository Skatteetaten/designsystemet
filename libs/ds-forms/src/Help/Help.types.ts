import { ReactElement, ReactNode } from 'react';

import { BaseProps } from '@skatteetaten/ds-core-utils';

export type OnHelpToggleHandler = (isOpen: boolean) => void;

export interface HelpProps {
  /** Skjuler hele komponenten, men er fortsatt synlig for skjermleser */
  hideHelp?: boolean;
  /** Hjelpetekst */
  helpText?: ReactNode;
  /** Overskriver default hjelpeikon */
  helpSvgPath?: ReactElement<SVGPathElement>;
  /** Overskriver default tooltip-tekst til hjelpeikon */
  titleHelpSvg?: string;
  /** Label eller legend id */
  targetId?: string;
  /** Beskrivelse */
  description?: string;
  /** Callback som kalles n√•r hjelpetekst vises/skjules */
  onHelpToggle?: OnHelpToggleHandler;
  className?: BaseProps['className'];
  classNames?: {
    helpText?: string;
    description?: string;
  };
}
