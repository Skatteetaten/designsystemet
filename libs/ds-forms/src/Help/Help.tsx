import { JSX, useRef, useState } from 'react';
import { useTranslation } from 'react-i18next';

import { IconButton } from '@skatteetaten/ds-buttons';
import { dsI18n } from '@skatteetaten/ds-core-utils';
import { CancelSVGpath, HelpSimpleSVGpath } from '@skatteetaten/ds-icons';

import { HelpProps } from './Help.types';

import styles from './Help.module.scss';

export const Help = ({
  description,
  helpText,
  helpSvgPath,
  hideHelp,
  targetId,
  titleHelpSvg,
  onHelpToggle,
  className,
  classNames,
}: HelpProps): JSX.Element => {
  const { t } = useTranslation('Shared', { i18n: dsI18n });

  const [showHelpText, setShowHelpText] = useState(false);
  const helpButtonRef = useRef<HTMLButtonElement>(null);

  const svgHelpIcon = helpSvgPath ?? HelpSimpleSVGpath;
  const titleHelpIcon = titleHelpSvg ?? t('shared.Help');

  const toggleHelpText = (): void => {
    onHelpToggle?.(!showHelpText);
    setShowHelpText(!showHelpText);
  };

  const closeHelpText = (): void => {
    setShowHelpText(false);
    helpButtonRef.current?.focus();
    onHelpToggle?.(false);
  };

  const marginTopClassName = showHelpText ? styles.descriptionMarginTop : '';
  const hideHelpClassName = hideHelp ? styles.srOnly : '';

  return (
    <>
      {helpText && (
        <IconButton
          ref={helpButtonRef}
          className={`${styles.helpButton} ${hideHelpClassName}`.trim()}
          svgPath={svgHelpIcon}
          title={titleHelpIcon}
          size={'extraSmall'}
          ariaExpanded={showHelpText}
          ariaDescribedby={targetId}
          isOutlined
          onClick={(): void => toggleHelpText()}
        />
      )}
      {helpText && showHelpText && (
        <div
          className={`${styles.helpBox} ${hideHelpClassName} ${
            className ?? ''
          } ${classNames?.helpText ?? ''}`.trim()}
        >
          <span className={styles.helpBoxText}>{helpText}</span>
          <IconButton
            className={styles.helpBoxCloseButton}
            size={'small'}
            svgPath={CancelSVGpath}
            title={t('shared.Close')}
            onClick={(): void => closeHelpText()}
          />
        </div>
      )}
      {description && (
        <div
          className={`${
            styles.description
          } ${marginTopClassName} ${hideHelpClassName} ${
            classNames?.description ?? ''
          }`.trim()}
        >
          {description}
        </div>
      )}
    </>
  );
};

Help.displayName = 'Help';
