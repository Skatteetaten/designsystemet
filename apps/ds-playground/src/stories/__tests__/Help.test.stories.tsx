import { dsI18n } from '@skatteetaten/ds-core-utils';
import { WarningSVGpath } from '@skatteetaten/ds-icons';
import { Meta, StoryObj } from '@storybook/react';
import { expect, fireEvent, within } from '@storybook/test';

// eslint-disable-next-line @nx/enforce-module-boundaries
import {
  loremIpsumWithoutSpaces,
  wrapper,
} from './testUtils/storybook.testing.utils';
// eslint-disable-next-line @nx/enforce-module-boundaries
import { Help } from '../../../../../libs/ds-forms/src/Help/Help';
import { SystemSVGPaths } from '../utils/icon.systems';

const meta = {
  component: Help,
  title: 'Tester/Help (intern)',
  argTypes: {
    // Props
    description: { table: { disable: true } },
    classNames: { table: { disable: true } },
    hideHelp: { table: { disable: true } },
    helpText: { table: { disable: true } },
    helpSvgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    titleHelpSvg: { table: { disable: true } },
    targetId: { table: { disable: true } },
    // Events
    onHelpToggle: { table: { disable: true } },
  },
} satisfies Meta<typeof Help>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultDescription = 'En liten beskrivelse';
const defaultHelpText =
  'Vi trenger Ã¥ vite navnet ditt dersom vi skal kontakte deg senere.';

export const Defaults = {
  name: 'Defaults',
  play: async ({ canvasElement }): Promise<void> => {
    // eslint-disable-next-line testing-library/no-node-access
    const helpButton = canvasElement.querySelector('button');
    await expect(helpButton).not.toBeInTheDocument();
  },
} satisfies Story;

export const WithHelptext = {
  name: 'With HelpText (FS-A1, FS-A5 delvis, FS-B2 delvis)',
  args: {
    helpText: defaultHelpText,
  },
  argTypes: {
    helpText: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      hover: `${wrapper} > button`,
      focus: `${wrapper} > button`,
      click: `${wrapper} > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button');
    await expect(helpButton).toBeInTheDocument();
    await expect(helpButton).toHaveAttribute('aria-expanded', 'false');
    const helpSvg = canvas.getByLabelText(dsI18n.t('Shared:shared.Help'), {
      selector: 'svg',
    });
    await expect(helpSvg).toBeInTheDocument();
  },
} satisfies Story;

export const WithDescription = {
  name: 'With Description (FS-A5 delvis)',
  args: {
    description: defaultDescription,
  },
  argTypes: {
    description: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const descriptionNode = canvas.getByText(defaultDescription);
    await expect(descriptionNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithHelpTextAndDescription = {
  name: 'With HelpText And Description (FS-A5 delvis)',
  args: {
    helpText: defaultHelpText,
    description: defaultDescription,
  },
  argTypes: {
    helpText: { table: { disable: false } },
    description: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} > button`,
      click: `${wrapper} > button`,
    },
  },
} satisfies Story;

export const WithLongHelpTextAndDescription = {
  name: 'With Long HelpText And Description',
  args: {
    helpText: loremIpsumWithoutSpaces,
    description: loremIpsumWithoutSpaces,
  },
  argTypes: {
    helpText: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
    imageSnapshot: {
      click: `${wrapper} > button`,
    },
  },
} satisfies Story;

export const WithHelpSvgPathTitleHelpSvgAndTargetId = {
  name: 'With HelpSvgPath TitleHelpSvg And TargetId (FS-A5 delvis)',
  args: {
    helpText: defaultHelpText,
    helpSvgPath: WarningSVGpath,
    titleHelpSvg: 'Tooltip',
    targetId: 'storybook-root',
  },
  argTypes: {
    helpSvgPath: { table: { disable: false } },
    titleHelpSvg: { table: { disable: false } },
    targetId: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button');
    await expect(helpButton).toHaveAttribute(
      'aria-describedby',
      'storybook-root'
    );
    const svgNode = canvas.getByLabelText('Tooltip', { selector: 'svg' });
    await expect(svgNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithHideHelp = {
  name: 'With HideHelp (FS-A7)',
  args: {
    helpText: defaultHelpText,
    description: defaultDescription,
    hideHelp: true,
  },
  argTypes: {
    hideHelp: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button');
    const descriptionNode = canvas.getByText(defaultDescription);
    await expect(helpButton).toBeInTheDocument();
    await expect(descriptionNode).toBeInTheDocument();
  },
} satisfies Story;

export const ClickHelpButton = {
  name: 'Click HelpButton On And Off (FS-A5 delvis, FS-B2 delvis)',
  args: {
    helpText: defaultHelpText,
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button');
    await fireEvent.click(helpButton);
    await expect(helpButton).toHaveAttribute('aria-expanded', 'true');
    const helpText = canvas.getByText(defaultHelpText);
    await expect(helpText).toBeInTheDocument();
    await fireEvent.click(helpButton);
    await expect(helpText).not.toBeInTheDocument();
    await expect(helpButton).toHaveAttribute('aria-expanded', 'false');
  },
} satisfies Story;

export const ClickCloseButton = {
  name: 'Close HelpText With CloseButton (FS-A5 delvis, FS-A6, FS-B2 delvis)',
  args: {
    helpText: defaultHelpText,
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button');
    await fireEvent.click(helpButton);
    const closeButton = canvas.getByTitle(dsI18n.t('Shared:shared.Close'));
    await expect(closeButton).toBeInTheDocument();
    const helpText = canvas.getByText(defaultHelpText);
    await fireEvent.click(closeButton);
    await expect(helpText).not.toBeInTheDocument();
    await expect(helpButton).toHaveFocus();
  },
} satisfies Story;

export const WithHelpToggleEvent = {
  name: 'With onHelpToggle Event',
  args: {
    helpText: 'Hjelpetekst',
    onHelpToggle: (isOpen: boolean): void => {
      alert(isOpen ? 'Hjelpetekst blir vist' : 'Hjelpetekst skjules');
    },
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
} satisfies Story;
