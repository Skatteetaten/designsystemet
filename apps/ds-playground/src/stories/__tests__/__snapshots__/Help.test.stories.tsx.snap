// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Help (intern) ClickCloseButton play-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:ra:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:ra:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Help (intern) ClickHelpButton play-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r8:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r8:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Help (intern) Defaults play-test 1`] = `
<div data-test-block="true">
</div>
`;

exports[`Tester/Help (intern) WithDescription play-test 1`] = `
<div data-test-block="true">
  <div class="_description_10ldw_17">
    En liten beskrivelse
  </div>
</div>
`;

exports[`Tester/Help (intern) WithHelpSvgPathTitleHelpSvgAndTargetId play-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-describedby="storybook-root"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r6:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r6:">
        Tooltip
      </title>
      <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Help (intern) WithHelpTextAndDescription smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r2:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r2:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
  <div class="_description_10ldw_17">
    En liten beskrivelse
  </div>
</div>
`;

exports[`Tester/Help (intern) WithHelpToggleEvent smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:rc:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:rc:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Help (intern) WithHelptext play-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r0:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r0:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Help (intern) WithHideHelp play-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27 _srOnly_10ldw_59"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r7:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r7:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
  <div class="_description_10ldw_17  _srOnly_10ldw_59">
    En liten beskrivelse
  </div>
</div>
`;

exports[`Tester/Help (intern) WithLongHelpTextAndDescription smoke-test 1`] = `
<div data-test-block="true">
  <button class="_button_g87u5_1 _button_extraSmall_g87u5_29 _button_outlined_g87u5_91 _helpButton_10ldw_27"
          type="button"
          aria-expanded="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_g87u5_22"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r4:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r4:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
  <div class="_description_10ldw_17">
    Loremipsumdolorsitamet.Allesomharlagetennettside,trengtlittfylltekstellerbaresurfetrundtp√•nettetharantageligvissettdisseordene,etterfulgtaventilsynelatendeeviglangtekstfyltmedlatinskeliksomsetninger.
  </div>
</div>
`;
